src/linux.cc

```diff
@@ -287,6 +292,58 @@ void update_gateway_info_failure(const char *reason) {
 /* Iface Destination Gateway Flags RefCnt Use Metric Mask MTU Window IRTT */
 #define RT_ENTRY_FORMAT "%63s %lx %lx %x %*d %*d %*d %lx %*d %*d %*d\n"
 
+int update_gateway_info2(const char *s) {
```

Comment:
@su8 Your favorite code smell. Prevents us from merging the other PR.

Are you addressing this and CodeFactor (duplicate code)? Do you think we should start at `0`?  Because the log prints `[]` for 0... and it would follow interface naming style too, eg `enp2s0` and `wlp3s0`.

Reply:
1 should be more accurate, as you said when you think of indexing you don't want to consider indexing 0 e.g  **lo**.

Reply:
Can we have `${iface}` print a list of interfaces separated by comma, I guess, to partially solve `#381`? The parameter `${if_existing FILE (string)}` can read off optional `(string)` should be able to find a partial match (`eno` ==> `eno1`) instead of a wildcard.

Reply:
Can you try this diff:

```diff
diff --git a/src/core.cc b/src/core.cc
index 31405de4..ed3ff6b7 100644
--- a/src/core.cc
+++ b/src/core.cc
@@ -1449,9 +1449,8 @@ struct text_object *construct_text_object(char *s, const char *arg, long line,
   END OBJ(gw_iface, &update_gateway_info) obj->callbacks.print =
       &print_gateway_iface;
   obj->callbacks.free = &free_gateway_info;
-  END OBJ(iface, 0) obj->data.s = STRNDUP_ARG;
-  update_gateway_info2(obj->data.s);
-  obj->callbacks.print = &print_gateway_iface;
+  END OBJ(iface, &update_gateway_info2) obj->data.s = STRNDUP_ARG;
+  obj->callbacks.print = &print_gateway_iface2;
   obj->callbacks.free = &gen_free_opaque;
   END OBJ(gw_ip, &update_gateway_info) obj->callbacks.print = &print_gateway_ip;
   obj->callbacks.free = &free_gateway_info;
diff --git a/src/linux.cc b/src/linux.cc
index c636b008..4965c924 100644
--- a/src/linux.cc
+++ b/src/linux.cc
@@ -292,7 +292,7 @@ void update_gateway_info_failure(const char *reason) {
 /* Iface Destination Gateway Flags RefCnt Use Metric Mask MTU Window IRTT */
 #define RT_ENTRY_FORMAT "%63s %lx %lx %x %*d %*d %*d %lx %*d %*d %*d\n"
 
-int update_gateway_info2(const char *s) {
+int update_gateway_info2(void) {
   FILE *fp;
   char iface[64];
   unsigned long dest;
@@ -321,24 +321,23 @@ int update_gateway_info2(const char *s) {
     }
     if (!(dest || mask) && ((flags & RTF_GATEWAY) || !gate)) {
       snprintf(e_iface, 49, "%s", iface);
-    }
-    if (0 == (strcmp(s, ""))) {continue;}
-    if (1U == x) {
-      snprintf(interfaces_arr[x++], 63, "%s", iface);
-      continue;
-    } else if (0 == strcmp(iface, interfaces_arr[x - 1])) {
-      continue;
-    }
-    for (z = 1; z < 63; z++) {
-      if (0 == strcmp(iface, interfaces_arr[z])) {
-        skip = 1;
-        break;
+      if (1U == x) {
+        snprintf(interfaces_arr[x++], 63, "%s", iface);
+        continue;
+      } else if (0 == strcmp(iface, interfaces_arr[x - 1])) {
+        continue;
       }
+      for (z = 1; z < 63; z++) {
+        if (0 == strcmp(iface, interfaces_arr[z])) {
+          skip = 1;
+          break;
+        }
+      }
+      if (0 == skip) {
+        snprintf(interfaces_arr[x++], 63, "%s", iface);
+      }
+      skip = 0;
     }
-    if (0 == skip) {
-      snprintf(interfaces_arr[x++], 63, "%s", iface);
-    }
-    skip = 0;
   }
   fclose(fp);
   return 0;
@@ -397,10 +396,27 @@ int gateway_exists(struct text_object *obj) {
 }
 
 void print_gateway_iface(struct text_object *obj, char *p, unsigned int p_max_size) {
+  (void)obj;
+  snprintf(p, p_max_size, "%s", gw_info.iface);
+
+}
+
+void print_gateway_iface2(struct text_object *obj, char *p, unsigned int p_max_size) {
   long int z = 0;
+  unsigned int x = 1;
+  char buf[64*64];
+  char *buf_ptr = buf;
 
-  if (!obj->data.s || (0 == (strcmp(obj->data.s, "")))) {
-    snprintf(p, p_max_size, "%s", gw_info.iface);
+  if (0 == strcmp(obj->data.s, "")) {
+    for (; x < 63; x++) {
+      if (0 == strcmp("", interfaces_arr[x])) {
+        break;
+      }
+      buf_ptr += snprintf(buf_ptr, 63, "%s, ", interfaces_arr[x]);
+    }
+    --buf_ptr;
+    *(--buf_ptr) = '\0';
+    snprintf(p, p_max_size, "%s", buf);
     return;
   }
 
diff --git a/src/linux.h b/src/linux.h
index b081ef0b..a50790f4 100644
--- a/src/linux.h
+++ b/src/linux.h
@@ -33,10 +33,11 @@ void print_ioscheduler(struct text_object *, char *, unsigned int);
 void print_laptop_mode(struct text_object *, char *, unsigned int);
 
 int update_gateway_info(void);
-int update_gateway_info2(const char *);
+int update_gateway_info2(void);
 void free_gateway_info(struct text_object *obj);
 int gateway_exists(struct text_object *);
 void print_gateway_iface(struct text_object *, char *, unsigned int);
+void print_gateway_iface2(struct text_object *, char *, unsigned int);
 void print_gateway_ip(struct text_object *, char *, unsigned int);
 
 enum { PB_BATT_STATUS, PB_BATT_PERCENT, PB_BATT_TIME };
```